doc update (#218)
curveball/a12n-server: docs/getting-started.md

To set up a new a12n-server from scratch, start by obtaining the following
prerequisites:

> 1. A working `nodejs` (node version needs to be > 14) and `npm` binary.

Reviewer: ```suggestion
1. NodeJS (version 14 or higher) and `npm`.
```

MySQL setup
-----------

After you have MySQL up and running, create a new empty database (/schema) & user for
`a12n-server`.

```sh
mysql> CREATE DATABASE a12nserver;
mysql> CREATE USER 'a12nserver' IDENTIFIED BY 'password';
```

> While running the application, it is possible to run into privileges issues.

Reviewer: This is incorrect. `GRANT` is _required_. This suggests that issues _might_ happen, but it should just be part of the earlier 2 steps.

You need to enable the GRANT statement to grant privileges and roles.

```sh
mysql> GRANT SELECT, INSERT, UPDATE, DELETE ON dbname.* TO 'a12nserver';
```

The next step is to insert the MySQL schemas that are shipping with the git
repository. The easiest is to just run:

```sh
cat mysql-schema/*.sql | mysql -u a12nserver -p -h hostname databasename
```

Once you have finalized the permissions that you want to set up for your new users,
always be sure to reload all the privileges.

```sh

> mysql> FLUSH PRIVILEGES;

Reviewer: I'd also just make this part of the earlier steps. It's just 4 mysql commands so I don't think there's a need for a lot of the 'flavor'

MySQL setup
-----------

> After you have MySQL up and running, create new empty database (/schema) & user for

Reviewer: I think `a` belongs in this sentence =)

`a12n-server`.

```sh
mysql> CREATE DATABASE db_name;

> mysql> CREATE USER 'username'@'localhost' IDENTIFIED BY 'password';

Reviewer: Might be a good idea to suggest `a12nserver` for both db and user, so it matches the defaults. Also mention that 'password' should be replaced by your own password.

Also drop the '@localhost` portion, and move up the GRANTS to the same section. No need to do this separate.

```sh
export MYSQL_PASSWORD=....
export MYSQL_USER=a12nserver
export MYSQL_DATABASE=a12nserver
docker run -it --rm -p 127.0.0.1:8531:8531 --name a12n-server-01 a12n-server
```

> Not docker:
> a12n-server comes with `dotenv` package. You can create `.env` file top
> of your project and use these environment variables instead.

Reviewer: How about:

```suggestion
If you are running a12nserver outside of docker, the easiest way to change environment variables is to create a `.env` file, and specify the following settings.
```

> ```sh

Reviewer: Might be good to mention `.env` for non-docker users...

UPDATE users SET active = 1 WHERE id = 1;
```

> You might not be an admin. Make yourself as an admin, run:

Reviewer: ```suggestion
To give your newly created user full admin privileges, run:
```

```
The SMTP_URL takes any format that that [Nodemailer](https://nodemailer.com/smtp/) takes.

Next steps
----------

> [Oauth2 Client Setup](https://github.com/curveball/a12n-server/blob/master/docs/oauth2-client.md)

Reviewer: Spelling is OAuth2 (uppercase A)
